### PR TITLE
Enable cudnn rnns when dropout is set

### PR DESCRIPTION
**Pull request by fchollet**

### CODE REVIEW

File: keras/src/layers/rnn/lstm.py [hunk 1]
```diff
@@ -518,14 +518,24 @@ def inner_loop(self, sequences, initial_state, mask, training=False):
             mask = mask[0]
 
         if self.use_cudnn in ("auto", True):
-            if not self.dropout and not self.recurrent_dropout:
+            if not self.recurrent_dropout:
                 try:
+                    if self.dropout:
```
Comment (miticollo) [inline keras/src/layers/rnn/lstm.py:523]: Shouldn't it be `if training and self.dropout`? Because we don't want to apply dropout during inference like happens when `self.use_cudnn` is `False`: https://github.com/keras-team/keras/blob/c356cae26e36ecd726dc1d5d41282dccdadf5f65/keras/src/layers/rnn/lstm.py#L274-L276
